### PR TITLE
refactor: Replace lazy_static with std::sync::LazyLock

### PR DESCRIPTION
**Pull request by Techassi**
`std::cell::LazyCell` and `std::sync::LazyLock` were stabilized in Rust 1.80.0. We can now use the standard library types instead of the `lazy_static` crate.

This change is marked as breaking, because it requires downstream operators to bump their Rust version to 1.80.0 to support the new stabilized standard library type.

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

File: crates/k8s-version/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Changed
+
+- Replace `lazy_static` with `std::cell::LazyCell` ([#xxx]).
+
+[#xxx](https://github.com/stackabletech/operator-rs/pull/xxx)
```
Comment (NickLarsenNZ) [inline crates/k8s-version/CHANGELOG.md:11]:
```suggestion
- Replace `lazy_static` with `std::cell::LazyCell` ([#827]).

[#827](https://github.com/stackabletech/operator-rs/pull/827)
```

File: crates/stackable-operator/src/kvp/key.rs [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-lazy_static! {
-    static ref KEY_PREFIX_REGEX: Regex =
-        Regex::new(r"^[a-zA-Z](\.?[a-zA-Z0-9-])*\.[a-zA-Z]{2,}\.?$").unwrap();
-    static ref KEY_NAME_REGEX: Regex =
-        Regex::new(r"^[a-z0-9A-Z]([a-z0-9A-Z-_.]*[a-z0-9A-Z]+)?$").unwrap();
-}
+// Lazily initialized regular expressions
+static KEY_PREFIX_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"^[a-zA-Z](\.?[a-zA-Z0-9-])*\.[a-zA-Z]{2,}\.?$")
+        .expect("failed to compile key prefix regex")
```
Comment (NickLarsenNZ) [inline crates/stackable-operator/src/kvp/key.rs:12]: praise: Much appreciated (`unwrap()` -> `expect()`)!

**Review by NickLarsenNZ — CHANGES_REQUESTED** [review]
Just the PR number needs fixing

File: crates/stackable-operator/src/kvp/key.rs (hunk 2 above)
Comment (Techassi) [reply crates/stackable-operator/src/kvp/key.rs:12]: :raised_hands:

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Review by sbernauer — APPROVED** [review]

**Pull request merged**